src/pybind/wrapper.cpp

```diff
@@ -78,26 +78,22 @@ std::tuple<std::vector<std::string>, std::string> call_solve_nonlinear_system(
                                  "vars"_a = vars,
                                  "function_calls"_a = function_calls);
     std::string script = R"(
-exception_message = ""
-try:
-    solutions = solve_non_lin_system(equation_strings,
+solutions = solve_non_lin_system(equation_strings,
                                      state_vars,
                                      vars,
                                      function_calls)
-except Exception as e:
-    # if we fail, fail silently and return empty string
-    solutions = [""]
-    new_local_vars = [""]
-    exception_message = str(e)
 )";
 
-    py::exec(nmodl::pybind_wrappers::ode_py + script, locals);
-    // returns a vector of solutions, i.e. new statements to add to block:
+    try {
+        py::exec(nmodl::pybind_wrappers::ode_py + script, locals);
+    } catch (py::error_already_set& e) {
```

Comment:
Including `ode.py` verbatim let exceptions escape (just don't install `sympy` and get mystery crashes) - this will catch everything and give meaningful errors. Plus the code looks a little cleaner.

Reply:
Can this be extracted into a separate issue? If there's an import error is must crash. If there's a different type when calling `solve_non_lin_system`, then the error signals "maybe we've done this one already, it's fine", leading to a warning. The latter is rather unfortunate and there's an issue here: https://github.com/BlueBrain/nmodl/issues/1259 but it's important to reduce the scope of what we catch as "meh, it's fine" as much as possible. This seems to wide the amount of things the catch all covers. Separate issue so it can be discussed properly please (without blocking here).

The previous code is more explicit about what it's doing. The `error_already_set` could easily be interpreted as

Reply:
Reverted in [5fb9b05](https://github.com/BlueBrain/nmodl/pull/1354/commits/5fb9b05e10b104ecfe0b3da5529aa81fc5ec28f5)

Reply:
Fully done in [79798ab](https://github.com/BlueBrain/nmodl/pull/1354/commits/79798abb3718cca7effdbb043af4ea872d7b124a)

Windows Git tools are really, really bad...